src/Text/Pandoc/Shared.hs

```diff
@@ -1028,6 +1031,19 @@ collapseFilePath = Posix.joinPath . reverse . foldl go [] . splitDirectories
     isSingleton _ = Nothing
     checkPathSeperator = fmap isPathSeparator . isSingleton
 
+--
+-- File selection from the archive
+--
+filteredFilesFromArchive :: Archive -> (FilePath -> Bool) -> [(FilePath, BL.ByteString)]
+filteredFilesFromArchive zf f =
+  mapMaybe (fileAndBinary zf) (filter f (filesInArchive zf))
+  where
+    fileAndBinary :: Archive -> FilePath -> Maybe (FilePath, BL.ByteString)
+    fileAndBinary a fp =
+      case findEntryByPath fp a of
```

Comment:
I'm no expert but I think it would be idiomatic to use the bind function `>>=` for the Maybe monad, here.
`findEntryByPath fp a >>= \e -> Just (fp, fromEntry e)`

See e.g. http://stackoverflow.com/a/10881861